### PR TITLE
review: chore: convert extra checks to github action

### PR DESCRIPTION
**Pull request by MartinWitt**

### CODE REVIEW

**Comment by MartinWitt** [comment]
We should really convert this to a github action.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -132,11 +131,80 @@ jobs:
...
+
+      - name: Check documentation links
+        run: python3 ./chore/check-links-in-doc.py
+
+      - name: Analyze dependencies through DepClean in spoon-core
+        run: mvn -q depclean:depclean
+
+      - name: Spoon-decompiler
+        run: |
+          cd spoon-decompiler
```
Comment (I-Al-Istannen) [inline .github/workflows/tests.yml:161]: How about using the `working-directory` step option for such things: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun?

**Review by I-Al-Istannen — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (MartinWitt) [reply .github/workflows/tests.yml]: I convert this. Also I remove the bash script

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -48,7 +48,7 @@ jobs:
       - uses: oracle-actions/setup-java@2c4df2930e35870536667f383d87f1246fbe613f # v1
         with:
           website: jdk.java.net
-          java-version: ${{ matrix.java }}
+          version: ${{ matrix.java }}
```
Comment (MartinWitt) [inline .github/workflows/tests.yml:51]: The oracle action uses version and not java-version.

**Review by SirYwell — COMMENTED** [review]
I think this is a great improvement to the CI. I have a few things that could be considered to further improve it (probably here, probably in separate PRs):

1. Most steps don't depend on each other. What do you think about having separate jobs for them?
2. We download some files (the z3 build, gradle). Can we make use of caching?
3.  It seems like most sub projects run similar maven commands, however, it's somewhat inconsistent (some run depclean, some run checkstyle, some run both). Is this something we want to improve?

File: .github/workflows/tests.yml (hunk 2 above)
Comment (MartinWitt) [reply .github/workflows/tests.yml]: Okay lets fix this in another PR.

**Comment by MartinWitt** [comment]
> I think this is a great improvement to the CI. I have a few things that could be considered to further improve it (probably here, probably in separate PRs):
> 
> 1. Most steps don't depend on each other. What do you think about having separate jobs for them?

Actually, most jobs need to install spoon-core/spoon-pom before. We dont want to run `mvn install` to often, as this takes time.

> 2. We download some files (the z3 build, gradle). Can we make use of caching?

Sure
> 3. It seems like most sub projects run similar maven commands, however, it's somewhat inconsistent (some run depclean, some run checkstyle, some run both). Is this something we want to improve?

I hope there is some reasoning behind this

**Comment by MartinWitt** [comment]
The z3 download is cached now.

**Review by I-Al-Istannen — APPROVED** [review]

**Pull request merged**